### ISSUE TITLE
Create a sorted integer index, enable dask divisions

### ISSUE THREAD

**Issue by dougbrn** [lincc-frameworks/tape#98]
Currently, lsstseries uses the ensemble._id_col column as the primary index. This makes intuitive sense for fast grouping on objects, joins between object and source, etc. However, one important consideration for scale is the ability to assign dask divisions to the partition structure. Divisions require a sorted integer index, and indicate which slices of index values live in each partition. This will save on time spent searching partitions for a given object id, as divisions provide a means to just access partitions that include the required information.

However, the sorted integer requirement is difficult to impose on the user chosen id column, there is no guarantee that the column is sorted or even an integer. Therefore, to support divisions we should demote self._id_col to be just another column in the source dataframe, and instead create a sorted integer index that maps to self._id_col.

This sorted integer index should be unique-per-object, just like the object_id. It should be generated in any loader function

Blocks #88 , a sorted integer index will allow for potential utilization of the hipscat_id and any hipscat loader function may instead generate the index directly from the association table

**Comment by dougbrn** [comment]
I've spent a few days working on a [branch](https://github.com/lincc-frameworks/lsstseries/tree/index_divisions) that would implement this, and ultimately failed to get any amount of performance boost out of it (and in fact things were several times slower. My sense is there's two main factors to this:

1. The operations involved with calculating and setting a new index are expensive in dask. My implementation approach was to generate an index off of the object table, join object to source, and reindex both tables on the new index. The reindex is expensive, especially when applied to the source table. It's tricky to get around this, as since this isn't a column that is available on read from parquet, we'll always have to set the index at some point. It seems to also be problematic for memory, as setting this index requires knowledge of the full id column to create the mapping. I ran out of memory when trying to apply this to larger datasets.
2. The instances where we actually utilize divisions are currently limited. Divisions should be helpful when Dask is looking for certain rows within the source table primarily. However, we only do this when performing a sync table operation, and even then only when we've modified the object table and are looking to join onto the source table. Our analysis functions at present only run on the information in the source table, doing groupbys.

These two suggest that the cons greatly outweigh the pros at this current stage. I will say that there is certainly the possibility that there is some way to implement this that could be useful, down the road or even now. I didn't want to go any further down the rabbit hole without some trust that we'd get some real benefit from this. Given all this, my suggestion is that we hold off on this for now and keep it in the back of our minds. 

As for how this relates to #88, I think the answer for now is that having lsstseries ingest the association information from hipscat is unlikely to actually improve the speed of common lsstseries workflows. However, we should eventually generalize lsstseries to use a sorted integer index if it's available. The hipscat id doesn't satisfy that for our purposes as it's unique per source and object, where we need something that is unique per source. But if the hipscat_id for the associated object were available in the source table that would be something we'd be motivated to use over the object ID potentially.
